quantgen/R/quantile_ensemble.R

```diff
@@ -120,17 +120,9 @@ quantile_ensemble = function(qarr, y, tau, weights=NULL,
   
   # Flexible stacking
   else {
-    # First properly set up q0, if there's noncrossing constraints 
-    if (noncross) {
-      # If there's no q0 passed, then just use the training points
-      if (is.null(q0)) q0 = qarr
-      # If there's one passed, then account for intercept if needed
-      if (!is.null(q0) && intercept) {
-        n0 = dim(q0)[1]; a0 = array(NA, dim=c(n0,p,r))
-        for (k in 1:r) a0[,,k] = cbind(rep(1,n0), q0[,,k])
-        q0 = a0
-      }
-    }
```

Comment:
Actually these lines need to stay because if there's `q0` passed and an intercept in the model then we need to append a column of 1s. But `if (!is.null(q0) && intercept)` should be changed to an `else if (...)`. That's why it was erroneously adding it twice.

Also, here also I think we will need `q0[,,k, drop = FALSE]`.

Reply:
Yes, this makes sense - so you'd need `else if (intercept)`, right?

Agree re `q0` except that I think it needs the same clunky `dim` syntax as above.

Reply:
Yes `else if (intercept)` looks right to me. Could you make that change on your branch here and I'll resolve and merge?